inject/android.js

```diff
@@ -16,10 +31,34 @@ function initCode () {
         platform: processedConfig.platform
     })
 
+    processedConfig.messageSecret = messageSecret
     init(processedConfig)
 
-    // Not supported:
-    // update(message)
+    // @ts-ignore
+    window.legacyContentScopeScriptsMessageCallback = (message) => {
```

Comment:
Not tested, but I suggest we harden this somewhat:

```suggestion
const secret = PROVIDED_BY_ANDROID
const secretMethodName = ALSO_PROVIDED_BY_ANDROID // Not containing above secret.

const wrappedUpdate = (providedSecret, ...args) => {
    if (providedSecret === secret) {
        update(...args);
    }
}.bind()

Object.defineProperty(window, secretMethodName, {
    value: wrappedUpdate
})

// Then Android code can call window[secretMethodName](secret, ...)
```

Notes:
 - `.bind()` prevents wrappedUpdate.toString() from displaying the function's contents.
 - Object.defineProperty (without specifying configurable or writable) won't allow the website to redefine the function.
 - Checking the secret should hopefully prevent the website calling update() itself.

Reply:
I've added these changes and it works as expected 👍